src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentMethodsScreen.tsx

```diff
@@ -235,6 +251,55 @@ const GenericError = () => {
   );
 };
 
+export const useOnRecurringPaymentReceived = ({
+  recurringPaymentId,
+  callback,
+}: {
+  recurringPaymentId: number | undefined;
+  callback: () => void;
+}) => {
+  const {userId} = useAuthContext();
+
+  const mapRecurringPaymentIds = (
+    d: FirebaseFirestoreTypes.DocumentSnapshot,
+  ): number => {
+    const recurringPayment = d.data();
+    if (!recurringPayment) {
+      throw new Error('No recurring payment data');
+    }
+
+    return recurringPayment.id;
+  };
+
+  const callbackRef = useCallback(() => callback(), [callback]);
```

Comment:
The naming is misleading since it is not a `ref`. And also the use of `useCallback` here doesn't achieve anything.

Take a look at the [`useCallback` doc](https://react.dev/reference/react/useCallback) to understand how it works and where it should be placed to achieve what we want.

Reply:
I think is best that you show here an example of how do you see it, more than I guessing what is the think you want to achieve, the useRef I used before I think works as it should.

Reply:
From the `useCallback` [doc](https://react.dev/reference/react/useCallback):
> On next renders, React will give you the same function again if the dependencies have not changed since the last render.

The `callback` isn't a stable reference as it is recreated on every render. And from the doc quoted above since the dependencies change on every render, that means the function returned by `useCallback` will also change on every render. This means using `useCallback` here has no effect.

Reply:
Yes, that is clear, but I'm not quite sure what your suggestion is for the code, because the callback does not have a parameter nor uses anything/dependencies, so we have a plain object callback that might not be accurate when it is in a dependency for useEffect, useCallback, you point out those things, that is fine.

So I guess your are suggesting then something like this:

```
const callbackRef = useRef(callback);
  useEffect(() => {
    callbackRef.current = callback;
 }, [callback]);
```

I have tried different ways now. So that is why I asked if you show here the code how do you see it then we can avoid to write here too much.

Reply:
Can make the `callback` function it self a stable reference by wrapping the creation of it with `useCallback`.

This is how the `callback` function is created now:
```ts
callback: async () => {
  closeInAppBrowser();
  await refetchRecurringPayment();
},
```

Reply:
Ok, I have tried that ;)